scripts/pre-commit/validate_changelog_update.py

```diff
@@ -0,0 +1,118 @@
+import subprocess
+import logging
+import re
+import sys
+import typing as tp
+from pathlib import Path
+
+logger = logging.getLogger("validate_changelog_update")
+logger.setLevel(logging.INFO)
+
+handler = logging.StreamHandler()
+handler.setLevel(logging.INFO)
+formatter = logging.Formatter("%(levelname)s - %(message)s")
+handler.setFormatter(formatter)
+logger.addHandler(handler)
+
+CODE_EXTENSIONS = {".json", ".sql"}
+VERSION_FILE = "VERSION.md"
+CHANGELOG_FILE = "CHANGELOG.md"
+
+
+class ValidationError(Exception):
+    """Custom exception for validation-related errors."""
+
+
+def is_code_file(filepath: str) -> bool:
+    return Path(filepath).suffix in CODE_EXTENSIONS
+
+
+def get_project_root() -> Path:
+    result = subprocess.run(
+        ["git", "rev-parse", "--show-toplevel"],
+        capture_output=True,
+        text=True,
+        check=True
+    )
+    return Path(result.stdout.strip())
+
+
+def read_file_content(file_path: Path) -> str:
+    if not file_path.is_file():
```

Comment:
This will return if its a file or directory? Do you want to check if it exists using `path.exists()` ?